src/hub.js

```diff
@@ -384,16 +384,17 @@ export async function getSceneUrlForHub(hub) {
     isLegacyBundle = !(glbAsset || hasExtension);
   }
 
-  if (isLegacyBundle) {
+  if (sceneUrl.indexOf("blob:") === 0) {
+    return document.querySelector("a-scene").is("entered") ? sceneUrl : loadingEnvironment;
+  } else if (isLegacyBundle) {
     // Deprecated
     const res = await fetch(sceneUrl);
     const data = await res.json();
     const baseURL = new URL(THREE.LoaderUtils.extractUrlBase(sceneUrl), window.location.href);
-    sceneUrl = new URL(data.assets[0].src, baseURL).href;
+    return new URL(data.assets[0].src, baseURL).href;
   } else {
-    sceneUrl = proxiedUrlFor(sceneUrl);
+    return proxiedUrlFor(sceneUrl);
```

Comment:
nit: I think a single return statement is generally better than multiple returns, but maybe it's more of a preference or me being pedantic. Feel free to ignore this if you feel otherwise.